ultimate-messenger/src/main/java/ru/progrm_jarvis/ultimatemessenger/format/model/AsmGeneratingTextModelFactory.java

```diff
@@ -0,0 +1,407 @@
+                                3 + TEXT_MODEL_JVM_CLASS_NAME_WITH_GENERIC_AND_DESCRIPTOR_LENGTH * dynamicElementCount
+                        ).append('(');
+                for (var i = 0; i < dynamicElementCount; i++) {
+                    descriptor.append(TEXT_MODEL_JVM_CLASS_NAME_WITH_DESCRIPTOR);
+                    signature.append(TEXT_MODEL_JVM_CLASS_NAME_WITH_GENERIC_AND_DESCRIPTOR);
+                }
+                // Add constructor `void <init>(TextModel[...TextModel])`
+                method = clazz.visitMethod(
+                        ACC_PUBLIC, CONSTRUCTOR_METHOD_NAME,
+                        descriptor.append(')').append('V').toString(),
+                        signature.append(')').append('V').toString(), null
+                );
+
+                method.visitCode();
+                //<editor-fold desc="Constructor code generation" defaultstate="collapsed">
+                // Get `this`
+                method.visitVarInsn(ALOAD, 0);
+                // Invoke Object constructor
+                method.visitMethodInsn(
+                        INVOKESPECIAL, OBJECT_JVM_CLASS_NAME, CONSTRUCTOR_METHOD_NAME, VOID_METHOD_SIGNATURE, false
+                );
+                // Set each field's value to constructor's parameter at the corresponding index
+                for (var i = 0; i < dynamicElementCount; i++) {
+                    method.visitVarInsn(ALOAD, 0);
+                    method.visitVarInsn(ALOAD, i + 1);
+                    method.visitFieldInsn(
+                            PUTFIELD, bytecodeClassName, GENERATED_FIELD_NAME_PREFIX + i,
+                            TEXT_MODEL_JVM_CLASS_NAME_WITH_DESCRIPTOR
+                    );
+                }
+                // Return from method
+                method.visitInsn(RETURN);
+                //</editor-fold>
+
+                method.visitMaxs(
+                        2 /* Always 2 for simple field-settings constructors */,
+                        dynamicElementCount + 1 /* {parameters} + [this] */
+                );
+                method.visitEnd();
+            }
+
+            val dynamicModels = new TextModel[dynamicElementCount];
+            { // Add `TextModel#getText(T)` method
+                // Implement `String getText(T)` method
+                method = clazz.visitMethod(
+                        ACC_PUBLIC, GET_TEXT_METHOD_NAME, GET_TEXT_METHOD_JVM_SIGNATURE,
+                        GET_TEXT_METHOD_JVM_GENERIC_SIGNATURE, null
+                );
+
+                method.visitCode();
+                //<editor-fold desc="Method code generation" defaultstate="collapsed">
+                method.visitTypeInsn(NEW, STRING_BUILDER_JVM_CLASS_NAME);
+                method.visitInsn(DUP);
+                // Specify initial length of StringBuilder via its constructor
+                {
+                    val length = staticLength;
+                    switch (length) {
+                        case -1: {
```

Comment:
Actually, the length cannot be less than 0.

Reply:
Resolved